### ISSUE TITLE
Add moderation contact information or button? (report content feature)

### ISSUE THREAD

**Issue by rugk** [PrivateBin/PrivateBin#674]
People continue to use our security email for reporting content on other PrivateBin instances. See https://github.com/PrivateBin/PrivateBin/issues/673 e.g.

Maybe we could thus:
* add a text at the bottom that is configurable with some contact information (mail or whatever an admin wants to set there)
* or even better, to avoid abuse of that: have a "report" button on a paste view that – for now – just displays these contact information, maybe with a mailto link/template for the reporter, so it automatically provides some useful information such as the URL, title and maybe the type of report this is ("spam", "personal information", "illegal information" etc.), which one could select before.

**Comment by bkil** [comment]
This should be highly prioritized.

Certain organized actors have started a campaign of repeatedly uploading a piece of illegal content on each instance and then promptly reporting it as abuse to the respective upstream hosting provider. Lawyers then threaten with legal action and disconnection of the whole server if the admin won't take down any and all illegal material within 1 hour of notice (at best, or just disconnect right away). Many admins are starting to decommission their instances as a result.

Such threat could be mitigated with a smart report button. It would only be shown on a decrypted view page. It could require solving a CAPTCHA and other bot protection and then providing a working email address. The reporting action would notify the admins immediately and hide the content for everybody within 1 hour unless reviewed by an admin.

If one of the admins happens to be awake, they could review the content and add its ID (and its content hash) to an allow list to immunize it against future automatic hiding due to bogus reports. If content is found to be abusive, its (perceptual, fuzzy) hash could be added to a (shared) blocklist and future hosting of the same could be denied (enforced client side by both the uploader and all viewers). The list could be bundled with the client or if it gets large, it might be represented in more efficient ways.

Until this is implemented, the default configuration should only allow expiry options up to 1 hour with a comment added next to it, so an unaware admin would be less impacted.

**Comment by rugk** [comment]
Any admin can configure the expiration policy they want.
I also think this feature would be useful, but how would it solve your problem? This needs more planning/ideas! Where would reports end up?

If admin also just get a notification mail then I guess this is no difference to what they get now.
So should this implement push notifications via https://ntfy.sh/ or so? (there was another bridge with multiple distribution channels but I don't currently recall it's name).  (Edit: or https://github.com/caronc/apprise)

**Comment by rugk** [comment]
To show the problem: this can also lead to more spam if the "reporting" would be made too easy. You can then overwhelm admin with fraudulent reports of it is just a push of a button, it essentially also is a risk to keep in mind.

**Comment by rugk** [comment]
I initially proposed just a mail button as an example. This is easy to implement and it still leaves "reporting burden" on the reporter, not (only) on the maintainer, who then needs to handle that mail.

**Comment by rugk** [comment]
> If content is found to be abusive, its (perceptual, fuzzy) hash could be added to a (shared) blocklist and future hosting of the same could be denied (enforced client side by both the uploader and all viewers).

And I'm strongly against implementing upload filters btw! Even if it were possible! They set a dangerous precedent, could be brute forced etc.
Also, due to encryption, likely impossible to do as it randomized even the same  intent. And that is good!

Note the slogan and stated Readme goals: admins do not know of content on their server until they get a link. Then they can react.

It still seems like you want to solve a social problem with technology here as far as I see. The actual problem in your use case is: A ISP/provider has way to fast requirements to allow for removal of "bad" content and gives to few time.
I guess the solution could be to choose a better isp then. Even a perfect reporting process and tooling could only do "so much" - if an admin is asleep the one hour will pass anyway.
Thats just my opinion.

But what is imho not discussable and goes against everything this projects stands for: **We will not implement upload filters!**

**Comment by rugk** [comment]
> The reporting action would notify the admins immediately and hide the content for everybody within 1 hour unless reviewed by an admin.

That is a good idea though, it could be done similar to how Discourse does it though - hide it automatically until and admin makes a decision of X users reported it. (not only one report though!) 
it should be optional, but could be helpful in case of spam waves or whatever?

But yeah, this could all have impacts and lead to more (false positive) reports etc.
And until now only you aka one user has reported this problem you appear to be talking of btw.

**Comment by bkil** [comment]
You can go through the instance list to verify that most have now either disabled uploads, tied them to burn upon reading (i.e., unusable as a pasting service), reduced expiry or have deactivated their instance altogether and replaced the page with a farewell message published recently.

**Comment by rugk** [comment]
Yeah but these are valid admin decisions.  Also, it's neither new nor a new phenomena as far as we know.

It's also unclear whether this feature would make a difference. Anyway, we of course always appreciate PRs, but personally, I would suggest a simple solution with an (optional) "report button" feature, first.
